modules/cf-core/src/models/state-channel.ts

```diff
@@ -369,12 +370,12 @@ export class StateChannel {
 
   public static createEmptyChannel(
     multisigAddress: string,
-    proxyFactoryAddress: string,
+    addresses: CriticalStateChannelAddresses,
     userNeuteredExtendedKeys: string[]
   ) {
     return new StateChannel(
       multisigAddress,
-      proxyFactoryAddress,
+      addresses,
       userNeuteredExtendedKeys,
       new Map<string, AppInstanceProposal>([]),
       new Map<string, AppInstance>(),
```

Comment:
If possible we should add a check here that the supplied multisig address is the expected one based on the critical state channel addresses

Reply:
I'm confused, check them against what? We're being given 2 critical addresses to add to the state (via the `addresses` param), what do we compare these addresses to? Every time we call createEmptyState channel, we're passing in addresses from the NetworkContext so comparing them against that seems redundant.

Reply:
the `multisig` is dependent on those critical addresses right?

you can easily pass in a `multisig` that is different from what would be generated, and i would rather just know it would fail in that case

now it would probably have to be async so that may be a pain, and if were only using this within the package then its probably fine